Editorial Revision (#1)
peppelinux/draft-demarco-status-attestations: draft-demarco-status-attestations.md

--- abstract

Status Attestation is signed objects that demonstrate the validity status of a
Digital Credential.
These attestations are ephemeral and periodically provided

> to Holders, that can present these to Verifiers along

Reviewer: @asharif1990 I used the defined terms after their definition, withing the dedicated section.

Abstract and introduction sections doesn't define nothing

I'm available to merge these but I'd like to ask you to get informed from other authors about this before merging

Author: Okay, you are right. In the abstract and intro, it is possible to avoid them. I wasn't familiar with IETF style. we can avoid the capitalized terms in the intro and abstract then

OpenID Connect Core [@OpenID.Core], the term "JSON Web Token (JWT)"
defined by JSON Web Token (JWT) {{RFC7519}}.

Credential Issuer:
: Entity that is responsible for the issuance of the Digital Credentials.

> The Issuer is responsible about the lifecycle of their issued Digital Credentials

Reviewer: here we don't have the definition of Digital Credentials for example

# Requirements

> The general requirements that MUST be met while implementing the Status Attestation are listed in this section.  

Reviewer: the MUST is listed in each requirement, there may be some requirement in the future with SHOULD. That's why I prefer to use the normative language within each point

Author: okay. then probably, I will re-write it in a more natural way such as: "The general requirements for the implementation of  Status Attestation are listed in this section."

~~~

To validate that the Wallet Instance is entitled to request its Status Attestation,
the following requirements MUST be satisfied:

> - The Credential Issuer verifies the signature of the `credential_pop` object using

Reviewer: ```suggestion
- The Credential Issuer MUST verify the signature of the `credential_pop` object using
```

| JOSE Payload | Description | Reference |
| --- | --- | --- |

> | **iss** | Wallet identifier (MUST be equal to JWK thumbprint of the key in `cnf` Claim of Wallet Instance Attestation). | {{RFC9126}}, {{RFC7519}} |

Reviewer: `MUST be equal to JWK thumbprint of the key in `cnf` Claim of Wallet Instance Attestation`

this is the italian impl profile, not of interest in the IETF OAuth WG, please remove. IT is also not compatible with the DE approach, while we have achieved a compromise that I want to respect in any place.

> | **iss** | Wallet identifier (MUST be equal to JWK thumbprint of the key in `cnf` Claim of Wallet Instance Attestation). | {{RFC9126}}, {{RFC7519}} |

Reviewer: ```suggestion
| **iss** | Wallet identifier. | {{RFC9126}}, {{RFC7519}} |
```

# Presenting Status Attestations

The Wallet Instance that provides the Status Attestations SHOULD be included in the
`vp_token` JSON array, as defined in [@OpenID4VP], the Status Attestation along with the
related Digital Credential.

The Verifier that receives a Digital Credential supporting the Status Attestation,
SHOULD:

- Decode and validate the Digital Credential;

> - check the presence of `status.status_attestation` in the Digital Credential. If true, the Verifier MUST:

Reviewer: ```suggestion
- check the presence of `status.status_attestation` in the Digital Credential. If true, the Verifier SHOULD:
```

related Digital Credential.

The Verifier that receive a Digital Credential supporting the Status Attestation,
MUST:

- Decode and validate the Digital Credential;

> - check the presence of `status.status_attestation` in the Digital Credential. If true, the Verifier MUST:

Reviewer: it is a SHOULD since a RP may check the revocation or not, depending by different factors.

there are some credentials that even if expired doesn't mean that are not usable, eg: age over.

then for RP's viewpoint it is a SHOULD and not a MUST

Author: Okay, this sounds reasonable to me. However, somehow we should include this inside the draft, otherwise having should would be strange if you agree.
